### PR TITLE
Quickly connect all articles in a snippet to an imported resource

### PR DESCRIPTION
**Pull request by elpoelma**
## Overview
This PR adds a utility button to the snippet-editor sidebar which allows you to quickly connect all non-connected articles in a snippet to an imported resource.
- Button is greyed out if all articles are connected
- The resource URI is formed as follows: `http://example.org/imported-decision-${snippetList.id}`

### Setup
None

### How to test/reproduce
- Start the frontend
- Open a snippet which contains articles that are not connected to an imported resource yet
- Click the connect button + ensure it greys out
- Ensure all articles are now connected through the `http://data.europa.eu/eli/ontology#has_part` predicate
- Save the snippet and go back to it's snippet-list edit page
- The imported resource should now correctly show up

### Challenges/Uncertainties
- As this is an internal tool, I didn't bother with translations.

### Checks PR readiness
- [x] UI: works on smaller screen sizes
- [x] UI: feedback for any loading/error states
- [x] Check cancel/go-back flows
- [x] Check database state correct when deleting/updating (especially regarding relationships)
- [x] changelog
- [x] npm lint
- [x] no new deprecations

### CODE REVIEW

File: app/templates/snippet-management/edit/edit-snippet.hbs [hunk 1]
```diff
@@ -84,6 +84,10 @@
     <:aside>
       {{#if this.editor}}
         <Sidebar as |Sidebar|>
+          <ConnectArticlesToImportedResource
+            @controller={{this.editor}}
+            @importedResource={{this.importedDecisionUri}}
```
Comment (piemonkey) [inline app/templates/snippet-management/edit/edit-snippet.hbs:89]: I don't think it works well to pass in a URI, as it ignores any existing imported resources in the document. This also means matching URIs, so if there are e.g. 3 articles without an imported resource link in the snippet, we get 3 separate imported resources with the same URI. The issue of multiple imported resources with the same URI doesn't cause any issues that I can find so far though.

File: app/components/connect-articles-to-imported-resource.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    }
+  }
+
+  get documentImportedResources() {
+    return this.controller.getDocumentAttribute(IMPORTED_RESOURCES_ATTR) ?? [];
+  }
+
+  isConnected(articleNode) {
+    return Boolean(
+      (articleNode.attrs.backlinks ?? []).find(
```
Comment (piemonkey) [inline app/components/connect-articles-to-imported-resource.js:60]: .some()?

File: app/components/connect-articles-to-imported-resource.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    const articleURIs = this.nonConnectedArticleURIs;
+    for (const uri of articleURIs) {
+      this.controller.doCommand(
+        addProperty({
+          resource: this.importedResource,
+          property: {
+            predicate: 'http://data.europa.eu/eli/ontology#has_part',
+            object: sayDataFactory.resourceNode(uri),
+          },
+          isNewImportedResource: !this.documentImportedResources.includes(
```
Comment (piemonkey) [inline app/components/connect-articles-to-imported-resource.js:46]: I'm not sure why this doesn't seem to be working in my tests. See my comment in edit-snippet.hbs .

**Review by piemonkey — COMMENTED** [review]
Seems to work, though I'm not sure about some of the handling of existing imported resources.

Testing this also makes me realise that we need to open any template using these snippet lists, open the 'manage snippet lists' modal and hit save, then map the imported resource to the decision. Doing this once tells me that we need to tweak the styling to not have the imported resource list extending off the screen or hidden behind the document controls.

File: app/components/connect-articles-to-imported-resource.js (hunk 3 above)
Comment (elpoelma) [reply app/components/connect-articles-to-imported-resource.js:46]: Fixed, used the wrong variable name

File: app/templates/snippet-management/edit/edit-snippet.hbs (hunk 1 above)
Comment (elpoelma) [reply app/templates/snippet-management/edit/edit-snippet.hbs:89]: Ok so:
- The issue with multiple imported resources with the same URI is now solved.
- I indeed did not take into account already-exisiting imported resources. But I don't know if we already added imported resources to the IV snippets on QA and PROD. Additionally, we want to make sure we use the same imported-resource URI across the whole snippet-list.

**Review by abeforgit — APPROVED** [review]

**Pull request merged**